src/index.ts

```diff
@@ -172,9 +172,14 @@ export const main = async (unresolvedConfiguration: CommandLineOptions) => {
           const isHandled = deputy.maybeAddReferencedExternalDependency(workspace, packageName);
           if (!isHandled) collector.addIssue({ type: 'unlisted', filePath: containingFilePath, symbol: specifier });
 
-          // Pattern: @local/package/file
+          // Pattern: self-reference/import
           const otherWorkspace = chief.findWorkspaceByPackageName(packageName);
-          if (otherWorkspace && specifier !== packageName && workspace !== otherWorkspace) {
+          if (workspace.pkgName === otherWorkspace?.pkgName) {
```

Comment:
`workspace === otherWorkspace`

Doesn't work. The 2 workspaces look identical but aren't the same object.